src/core/server/ui_settings/ui_settings_service.ts

```diff
@@ -134,4 +137,17 @@ export class UiSettingsService
       }
     }
   }
+
+  private validateAndUpdateConfiguredDefaults(defaults: Record<string, any> = {}) {
+    for (const [key, value] of Object.entries(defaults)) {
+      const definition = this.uiSettingsDefaults.get(key);
+      if (!definition)
+        throw new Error(`[ui settings defaults [${key}]: expected key to be have been registered`);
```

Comment:
What's the impact of throwing here? Does it prevent the service starting? Is it presented to the user?

Reply:
It prevents OSD from starting. The other 2 validations as well as the other registration in this service do the same.

Reply:
Is that desirable? The error message is actually a bit cryptic if that's the case. The fact that it hasn't been registered is not really the issue the admin needs to fix, right? Isn't the mitigation to update the config to remove `defaults` from a key that doesn't support them?

Reply:
I agree; they make no sense. This error is a rewording of the existing one from here:
https://github.com/opensearch-project/OpenSearch-Dashboards/blob/eede3c15f9860a8a197cf7531e20ed356bd525be/src/core/server/ui_settings/ui_settings_service.ts#L118